2022-OCT/answer.md

```diff
@@ -0,0 +1,149 @@
+<!-- @import "[TOC]" {cmd="toc" depthFrom=1 depthTo=6 orderedList=false} -->
+
+# AACS3013 Oct 2022 Answers
+
+[Link to the paper](https://eprints.tarc.edu.my/23328/1/AACS3013.pdf)
+
+- [Question 1](#question-1)
+- [Question 2](#question-2)
+- [Question 3](#question-3)
+- [Question 4](#question-4)
+
+## Answers
+
+### Question 1
+
+a) i)
+
+Primary key is a unique identifier for each record in a database table. For instance, in a library management system, we will have a BOOKS table. In the BOOKS table, it contains BookID as a primary key to ensure that no two books can have the same identifier.
+
+a) ii)
+
+For example, a university course enrollment system contains 'Students' table, 'Courses' table and 'Enrollments' table that manages the many to many relationship between students and courses for each semester. Therefore, the 'Enrollments' table will contain three composite keys namely StudentID, CourseID and Semester. The issue with composite key is that other tables that need to reference this 'Enrollments' table must include all parts of the composite key (StudentID, CourseID and Semester) which complicates the foreign key relationships. To improve this design, we can use surrogate key for the 'Enrollments' table namely EnrollmentID. WHen other tables want to reference this 'Enrollments' table, they only need to include the single 'EnrollmentID' column which simplifies foreign key relationships.
+
+b) 
+
+DBMS Functions : 
+- Backup and Recovery management provides backup and data recovery procedures.
+- Data dictionary management defines data elements and their relationships
+- Data storage management stores data and related data entry forms, report definitions etc.
+
+c)
+
+- Data redundancy is when different and conflicting versions of the same data appear in different places.
```

Comment:
This is more like the definition of data inconsistency.

Data redundancy occurs when data of the same object is repeatedly defined multiple times within a single database. The redundancy creates space inefficiency and may produce data inconsistency.